README.md

```diff
@@ -49,85 +49,105 @@ The included Spring Boot Auto-configuration will enable the htmx integrations.
+The first is to use annotations, e.g. `@HxTrigger`, and the second is to use the class [HtmxResponse](https://javadoc.io/doc/io.github.wimdeblauwe/htmx-spring-boot/latest/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.html) as the return type of the controller method.
+
+See [Response Headers Reference](https://htmx.org/reference/#response_headers) for the related htmx documentation.
+
+The following annotations are currently supported:
+* [@HxRefresh](https://javadoc.io/doc/io.github.wimdeblauwe/htmx-spring-boot/latest/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxRefresh.html)
+* [@HxTrigger](https://javadoc.io/doc/io.github.wimdeblauwe/htmx-spring-boot/latest/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxTrigger.html)
+
+>**Note** Please refer to the related Javadoc to learn more about the available options.
 
-Setting the `hx-trigger` header triggers an event when the response is swapped in by htmx.
-The `@HxTrigger` annotation supports doing that for you:
+#### Examples
+
+If you want htmx to trigger an event after the response is processed, you can use the annotation `@HxTrigger` which sets the necessary response header [HX-Trigger](https://htmx.org/headers/hx-trigger/).
 
 ```java
-@GetMapping("/users")
 @HxRequest
-@HxTrigger("userUpdated") // 'userUpdated' event will be triggered by htmx
+@HxTrigger("userUpdated") // the event 'userUpdated' will be triggered by htmx
+@GetMapping("/users")
 public String hxUpdateUser(){
-    return "users";
+    return "partial";
+}
+```
+
+If you want to do the same, but in a more flexible way, you can use `HtmxResponse` as the return type in the controller method instead.
+```java
+@HxRequest
+@GetMapping("/users")
+public HtmxResponse hxUpdateUser(){
+    return new HtmxResponse()
+        .addTrigger("userUpdated") // the event 'userUpdated' will be triggered by htmx
+        .addTemplate("partial");
 }
 ```
 
 ### OOB Swap support
 
-htmx supports updating multiple targets by returning multiple partials in a single response with
-[`hx-swap-oob`](https://htmx.org/docs/#oob_swaps). Return partials using this library use the `HtmxResponse` as a return
-type:
+htmx supports updating multiple targets by returning multiple partials in a single response, which is called [Out Of Band Swaps](https://htmx.org/docs/#oob_swaps).
+For this purpose, use [HtmxResponse](https://javadoc.io/doc/io.github.wimdeblauwe/htmx-spring-boot/latest/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.html)
+as the return type of a controller method, where you can add multiple templates.
 
 ```java
+@HxRequest
 @GetMapping("/partials/main-and-partial")
 public HtmxResponse getMainAndPartial(Model model){
     model.addAttribute("userCount", 5);
     return new HtmxResponse()
-        .addTemplate("users :: list")
-        .addTemplate("users :: count");
+        .addTemplate("user-list")
+        .addTemplate("user-count");
```

Comment:
Why the removal of the `::` notation? That is a common usage with partials. (At least in my experience) I recommend keeping them since it can be a hiccup for folks new to Thymeleaf.

Reply:
Because they are Thymeleaf specific and this part of the documentation is no longer Thymeleaf related. It could be confusing for users like me who use another template engine, e.g. FreeMarker.

Reply:
I understand your point @xhaggi , but I would like to keep such an example in the docs for those of that do use Thymeleaf so they know  a partial fragment can be used there. Not sure if we can somehow combine the examples or just add a 2nd example, indicating that it is Thymeleaf specific?

Reply:
I have added a new sub-section "Markup Selectors and Out Of Band Swaps" under the "Thymleaf" section explaining this in more detail.